src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/RequestPayloadValueResolver.php

```diff
@@ -202,4 +216,21 @@ private function mapRequestPayload(Request $request, string $type, MapRequestPay
             throw new BadRequestHttpException(sprintf('Request payload contains invalid "%s" property.', $e->property), $e);
         }
     }
+
+    private function getController(Request $request): string
```

Comment:
I was looking at class ArgumentResolver which has a getPrettyName method, and I'm wondering if we couldn't leverage it: here, instead of a LogicException, we could throw a NearMissValueResolverException, with a shorter message:

Added by ArgumentResolver:
>Controller "%s" requires the "$%s" argument that could not be resolved.
then:
- Please set the $type argument of the #[MapRequestPayload] attribute to the type of the objects in the expected array.
- Please set the type to "array" type when using argument $type of #[MapRequestPayload]

Reply:
Addressed according your suggestions.